ports/broadcom/common-hal/busio/I2C.c

```diff
@@ -35,35 +35,63 @@
 #include "peripherals/broadcom/cpu.h"
 #include "peripherals/broadcom/vcmailbox.h"
 
-#define NUM_I2C (2)
+#if BCM_VERSION == 2711
+#define NUM_I2C (8)
+STATIC BSC0_Type *i2c[NUM_I2C] = {BSC0, BSC1, NULL, BSC3, BSC4, BSC5, BSC6, NULL};
+#else
+#define NUM_I2C (3)
+STATIC BSC0_Type *i2c[NUM_I2C] = {BSC0, BSC1, NULL};
+#endif
 
 STATIC bool never_reset_i2c[NUM_I2C];
-STATIC bool in_use_i2c[NUM_I2C];
-STATIC BSC0_Type *i2c[2] = {BSC0, BSC1};
+STATIC bool i2c_in_use[NUM_I2C];
 
 void reset_i2c(void) {
+    // BSC2 is dedicated to the first HDMI output.
+    never_reset_i2c[2] = true;
+    i2c_in_use[2] = true;
+    #if BCM_VERSION == 2711
+    // BSC7 is dedicated to the second HDMI output.
+    never_reset_i2c[7] = true;
+    i2c_in_use[7] = true;
+    #endif
     for (size_t i = 0; i < 2; i++) {
```

Comment:
Should this be `NUM_I2C`? If `2` is right, why `2`?

Reply:
Yup, it should. Good eye!